### PR TITLE
Adjust JSONEditor state handling

### PR DESCRIPTION
**Pull request by orlandohohmeier**
Adjust the JSONEditor state handling to ensure the editor content properly updates if the internal data is different then the input data, even when the input data is the same as the initial input data. This change fixes an issue in which the JSONEditor didn't update if the input value was the same as the initial value.

### CODE REVIEW

**Comment by wavesoft** [comment]
Just Beautiful

File: src/js/components/JSONEditor.js [hunk 1]
```diff
@@ -423,8 +430,8 @@ class JSONEditor extends React.Component {
       //
       // If nothing is found, default to root ([])
       //
-      const candidates = this.jsonMeta.reduce((memo, token) => {
-        const isMatch = token.path.every((component, i) => {
+      const candidates = this.jsonMeta.reduce(function (memo, token) {
```
Comment (nLight) [inline src/js/components/JSONEditor.js]: I thought we don't do cowboy style :trollface:

**Comment by nLight** [comment]
feel free to merge it!

File: src/js/components/JSONEditor.js [hunk 2]
```diff
@@ -304,17 +309,15 @@ class JSONEditor extends React.Component {
...
-          const cursor = JSONEditorUtil.cursorFromOffset(
-            parseInt(offset),
-            jsonText
-          );
-
-          return `at line ${cursor.row}:${cursor.column}`;
-        }
-      );
-    };
+      jsonError = errorStr.replace(/at position (\d+)/g, (match, offset) => {
```
Comment (orlandohohmeier) [inline src/js/components/JSONEditor.js]: ✔️  Done

**Pull request merged**